lib/DDGC/GitHub.pm

```diff
@@ -8,25 +8,35 @@ use DateTime::Format::ISO8601;
 use DateTime::Duration;
 use HTTP::Request;
 use JSON::MaybeXS;
+use DDP;
 
-has ddgc => (
-  isa => 'DDGC',
-  is => 'ro',
-  weak_ref => 1,
-  required => 1,
-);
 
-sub gh { DDGC::GitHub::Cmds->new($_[0]->net_github->args_to_pass) }
+=head1 SYNOPSIS
 
-has net_github => (
-  is => 'ro',
-  lazy_build => 1,
-);
+    use DDGC::GitHub;
+
+    my $github = DDGC::GitHub->new(ddgc => $ddgc);
+
+    # update all the github_* databases
+    $github->update_database;
+
+    # something to do with users giving us access to their github accounts via oath
+    $github->validate_session_code($code, $user);
+
+    # find or update github_user table for this user
+    $github->find_or_update_user($login);
+
+=cut
+
+has ddgc       => (isa => 'DDGC', is => 'ro', weak_ref => 1, required => 1);
```

Comment:
Yeah, DDGC::Config is usually accessed from the DDGC instance. So, having DDGC instances passed into application components doesn't have to be a problem, since the application components live as long as the application so circular refs don't really matter. We do end up with leaks elsewhere because of DBIC abuse.

In this case, I would just pass a `ddgc` ref as that happens to be easiest - you then get ->config and ->rs.

My problem with this generally is the separation of concerns gets really fuzzy, for example a template can send email by calling methods of `$c.d.postman` - this has led to a tendency where chunks of functionality exist across all layers rather than, say, adhering to fat model. It has also earned us some pretty bad leaks.

For now, having a weak ref to `ddgc` is how DDGC is designed to work so go with the flow :smiley: